comp/trace/agent/run.go

```diff
@@ -49,12 +48,6 @@ func runAgentSidekicks(ag *agent) error {
 		log.Warnf("Can't setup core dumps: %v, core dumps might not be available after a crash", err)
 	}
 
-	err = manager.ConfigureAutoExit(ag.ctx, coreconfig.Datadog)
-	if err != nil {
-		ag.telemetryCollector.SendStartupError(telemetry.CantSetupAutoExit, err)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Maybe I'm wrong but with this new implementation it will not call
`ag.telemetryCollector.SendStartupError(telemetry.CantSetupAutoExit, err)` if autoExit component fails to initialize ? How do you handle it ?
Isn't it a problem for Trace Agent team ?

Reply:
I would let the @DataDog/agent-apm team answer that question. I'm not familiar enough with this part of the code. it might allow us to simplify and get rid of `telemetry.CantSetupAutoExit` event

Reply:
I would say that we likely want to keep this to a "refactor", so we should ensure (either with an integration / e2e test / or manual test) that this behavior remains the same. ie. that if the component fails to initialize that the appropriate telemetry event is emitted, and that the trace-agent does not start. I took a look for any examples in the existing code base that test this behavior as it exists today and unfortunately I didn't see any.